src/pad_controller.rs

```diff
@@ -0,0 +1,28 @@
+// Copyright 2013-2018, The Gtk-rs Project Developers.
+// See the COPYRIGHT file at the top-level directory of this distribution.
+// Licensed under the MIT license, see the LICENSE file or <http://opensource.org/licenses/MIT>
+
+use ffi;
+use glib::IsA;
+use glib::translate::*;
+
+use PadActionEntry;
+use PadController;
+
+pub trait PadControllerExtManual {
+    #[cfg(any(feature = "v3_22", feature = "dox"))]
+    fn set_action_entries(&self, entries: &[&PadActionEntry]);
+}
+
+impl<O: IsA<PadController>> PadControllerExtManual for O {
+    #[cfg(any(feature = "v3_22", feature = "dox"))]
+    fn set_action_entries(&self, entries: &[&PadActionEntry]) {
+        let mut entries = entries.iter().map(|p| p.to_glib()).collect::<Vec<_>>();
```

Comment:
If you add a `[repr(C)]` to `PadActionEntry`, you can simply to `entries.as_ptr()` here instead of the copy

Reply:
Seems weird to convert a `&[&T]` into a `*mut T`. There is an extra pointer level no?

Reply:
Or should I just transform `&[&T]` parameter to `&[T]`? Sounds simpler.